third-culture-software/bhima
# Prevent selection of expired lots by the user in Stock-Exit (except for Loss)

For Stock-Exit to Patient, Service, or Depot, once the user has selected the inventory item, the drop-down menu to select the lot/batch will not show expired lots.  We also have a secondary check in the main submit function in the hopefully improbable case that expired lots come in via Invoices(Patient) or Requisitions(Service/Depot).

Suggested testing procedure using Bhima_test
 - ~Update the expiration date for Stock Lots for Quinine to make sure that one of the lots is not expired~.  <br>[This is no longer necessary due to the automatic updating of bhima_test data in PR https://github.com/IMA-WorldHealth/bhima/pull/4812]
 - Do Stock/Stock Exit, select Patient, Depot, or Service
 - Select the stock item for Quinine
 - Verify that only the unexpired lots are present in the Lots/Batches drop-down menu
 - Try the same procedure with Loss, the drop-down should show all the available lots including the expired ones.

Verify the French/English translations for the new error message:  ER_EXPIRED_STOCK_LOTS
This error message will be displayed as an error message (in red) at the top of the Stock Exit page when expired lots are found.

Note: There is no easy way to test for the error detection in the main submit button.  However if you modify `client/src/modules/stock/exit/exit.js` line 242 to be a copy of line 240 (temporarily in your local test sandbox), you should be able to test this by selecting an expired item (eg, in Patient) and then trying to Submit. You should see this error message in red at the top of the screen and the submit should fail.
